### PR TITLE
remove nested scaffolds

### PR DESCRIPTION
**Pull request by Leptopoda**
From the flutter [docs](https://api.flutter.dev/flutter/material/Scaffold-class.html)
>Nested Scaffolds
>
>The Scaffold is designed to be a top level container for a MaterialApp. This means that adding a Scaffold to each route on a Material app will provide the app with Material's basic visual layout structure.
>
>It is typically not necessary to nest Scaffolds. For example, in a tabbed UI, where the bottomNavigationBar is a TabBar and the body is a TabBarView, you might be tempted to make each tab bar view a scaffold with a differently titled AppBar. Rather, it would be better to add a listener to the TabController that updates the AppBar

We might also need this when tackling #282 and implement `Scaffold.of(context).hasFloatingActionButton`.
Now every `*Page` will have one and only one scaffold. Other widgets won't use Scaffold.

Another nice benefit of this is switching between a view that has a fab and one that doesn't (like `NotesView` and `NotesCategoriesView`) the fab will automatically animate in and out as needed.

---

It's less than the diff might suggest. Removing a widget will change the formatting quite a bit.

### CODE REVIEW

File: packages/neon/neon_news/lib/widgets/feed_fab.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+part of '../neon_news.dart';
+
+class NewsFeedFAB extends StatelessWidget {
```
Comment (provokateurin) [inline packages/neon/neon_news/lib/widgets/feed_fab.dart:3]: I'm not sure if abbreviating FloatingActionButton is a good idea or not. On the one hand it's a commonly used abbreviation that will be understood by most people, on the other hand we don't abbreviate any other widget name. I would rather spell it out completely, but we can also keep it this way :woman_shrugging:

File: packages/neon/neon_files/lib/widgets/browser_view.dart [hunk 2]
```diff
@@ -6,7 +6,6 @@ class FilesBrowserView extends StatefulWidget {
     required this.filesBloc,
     this.onPickFile,
     this.enableFileActions = true,
-    this.enableCreateActions = true,
```
Comment (provokateurin) [inline packages/neon/neon_files/lib/widgets/browser_view.dart:9]: Nice cleanup!

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/neon/neon/lib/src/pages/home.dart [hunk 3]
```diff
@@ -96,6 +96,8 @@ class _HomePageState extends State<HomePage> {
       }
     });
 
+    registerGlobalPopups(context);
```
[the remark this answers is not in the document]
Comment (Leptopoda) [reply packages/neon/neon/lib/src/pages/home.dart:99]: That's bad :thinking:

[the remark this answers is not in the document]
Comment (Leptopoda) [reply packages/neon/neon/lib/src/pages/home.dart:99]: Ok we could add our own [ScaffoldMessenger](https://api.flutter.dev/flutter/material/ScaffoldMessenger-class.html).

Or we could rely on the one provided to us by MaterialApp (didn't know it will do so).
I'll probably have a look at it tomorrow,

I always thought that we need to run this inside a scaffold but that reminds me of the Scaffold.of(context) to ScaffoldMessenger.of(context) migration from a year or so ago. Makes more sense now.

[the remark this answers is not in the document]
Comment (Leptopoda) [reply packages/neon/neon/lib/src/pages/home.dart:99]: I still thought about it yesterday and this wouldn't really work.
While we have a parent ScaffoldMessenger the messenger itself wouldn't have a scaffold descendant yet thus failing.

Maybe just adding a delay. but that`s ugly.

File: packages/neon/neon_news/lib/widgets/feed_icon.dart
Comment (provokateurin) [inline packages/neon/neon_news/lib/widgets/feed_icon.dart:1]: Already separate commit, but should be separate PR :)

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Comment by Leptopoda** [comment]
I'll probably change a lot of this again with the AdaptiveLayout but it's a nice cleanup for now and we already have the PR so :woman_shrugging:

**Comment by provokateurin** [comment]
@Leptopoda Can you rebase?

**Review by provokateurin — APPROVED** [review]
